src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs

```diff
@@ -91,6 +98,17 @@ private ProcessArgumentBuilder GetArguments(FilePath solution, MSBuildSettings s
             return builder;
         }
 
+        private static string GetPlatformName(PlatformTarget platform)
+        {
+            switch (platform)
+            {
+                case PlatformTarget.MSIL:
+                    return "AnyCPU";                                    
+                default:
```

Comment:
I think it would be good to handle the different cases explicitly and throw a `InvalidOperationException` here.